packages/loopring_v3/contracts/amm/libamm/AmmJoinProcess.sol

```diff
@@ -162,6 +164,8 @@ library AmmJoinProcess
         amounts = new uint96[](ctx.tokens.length);
 
         if (ctx.totalSupply == 0) {
+            // Set virtual balances
+            ctx.vTokenBalancesL2 = ctx.settings.controller.getInitialVirtualBalances(join.joinAmounts);
```

Comment:
On line 177, `ctx.tokenBalancesL2[i]` may not represent the real actual balances in this pool - some token may be withdrawn to L1 for DeFi. 

The real total actual balances should be **ctx.tokenBalancesL2[i] + tokens in Defi**